### PR TITLE
Add membership status column to UserGroup

### PR DESCRIPTION
**Pull request by bduffany**

### CODE REVIEW

File: server/tables/tables.go [hunk 1]
```diff
@@ -12,6 +12,12 @@ import (
 
 const (
 	mySQLDialect = "mysql"
+
+	
+	// UserGroup membership status constants
+
+	MemberStatus = 0
+	MembershipRequestedStatus = 1
```
Comment (tylerwilliams) [inline server/tables/tables.go:20]: Can we make this a proto enum instead? I think we should try to keep go-specific stuff out of here and just stick to primitive data types.

**Review by tylerwilliams — CHANGES_REQUESTED** [review]

File: server/tables/tables.go (hunk 1 above)
Comment (bduffany) [reply server/tables/tables.go]: Done

File: proto/grp.proto [hunk 2]
```diff
@@ -99,3 +99,10 @@ message UpdateGroupRequest {
 }
 
 message UpdateGroupResponse {}
+
+enum GroupMembershipStatus {
```
Comment (tylerwilliams) [inline proto/grp.proto:103]: nit: add an unknown state as 0

enum GroupMembershipStatus {
  UNKNOWN_MEMBERSHIP_STATUS = 0;

  // The user is a member of the group.
  MEMBER = 1;

  // The user has requested to join the group but is not yet a member.
  REQUESTED = 2;
}

**Review by tylerwilliams — CHANGES_REQUESTED** [review]
LGTM (modulo one change -- please fix before submitting)

**Review by tylerwilliams — APPROVED** [review]

**Pull request merged**